### PR TITLE
fix: adjust the execution priority of request-id to fix opentelemetry has no request id

### LINKED ISSUE #7239: help request: 插件 request-id 与 opentelemetry 结合
### 问题
[request-id](https://apisix.apache.org/zh/docs/apisix/plugins/request-id) 支持三种方式生成: ["uuid", "snowflake", "nanoid"]
1. 我想使用 request-id  作为 [opentelemetry](https://apisix.apache.org/zh/docs/apisix/plugins/opentelemetry) 的 trace_id 但是 trace_id 需要符合 `[0-9a-f]{32}` 规范，request-id 并没有这样的生成器，是否可以新增一个 request-id 的生成方式？
2. 我想在 opentelemetry 的 additional_attributes 追加 request-id 属性，但是我试了很多种读取的方式，好像都不能读取，以下是我的配置，请问有啥方式可以读取？
```json
{
    "id": "1",
    "create_time": 1655032228,
    "update_time": 1655045468,
    "plugins": {
        "opentelemetry": {
            "additional_attributes": [
                "route_id",
                "x-request-id",
                "request-id",
                "http_x-request-id",
                "X-Request-Id",
                "http_X-Request-Id"
            ],
            "disable": false,
            "sampler": {
                "name": "always_on",
                "options": {
                    "fraction": 1,
                    "root": {
                        "name": "always_on"
                    }
                }
            }
        },
        "request-id": {
            "algorithm": "snowflake",
            "disable": false,
            "header_name": "X-Request-Id",
            "include_in_response": true
        }
    }
}
```

### 使用版本
官方`docker`镜像: `apache/apisix:2.14.1-centos`
```shell
docker pull apache/apisix:2.14.1-centos
````
- APISIX version (run `apisix version`): 
```text
/usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua version
2.14.1
```
- Operating system (run `uname -a`):
```text
Linux apisix1 5.10.104-linuxkit #1 SMP Thu Mar 17 17:08:06 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
```text
nginx version: openresty/1.19.9.1
built by gcc 9.3.1 20200408 (Red Hat 9.3.1-2) (GCC)
built with OpenSSL 1.1.1n  15 Mar 2022
TLS SNI support enabled
configure arguments: --prefix=/usr/local/openresty/nginx --with-cc-opt='-O2 -DAPISIX_BASE_VER=1.19.9.1.6 -DNGX_LUA_ABORT_AT_PANIC -I/usr/local/openresty/zlib/include -I/usr/local/openresty/pcre/include -I/usr/local/openresty/openssl111/include' --add-module=../ngx_devel_kit-0.3.1 --add-module=../echo-nginx-module-0.62 --add-module=../xss-nginx-module-0.06 --add-module=../ngx_coolkit-0.2 --add-module=../set-misc-nginx-module-0.32 --add-module=../form-input-nginx-module-0.12 --add-module=../encrypted-session-nginx-module-0.08 --add-module=../srcache-nginx-module-0.32 --add-module=../ngx_lua-0.10.20 --add-module=../ngx_lua_upstream-0.07 --add-module=../headers-more-nginx-module-0.33 --add-module=../array-var-nginx-module-0.05 --add-module=../memc-nginx-module-0.19 --add-module=../redis2-nginx-module-0.15 --add-module=../redis-nginx-module-0.3.7 --add-module=../ngx_stream_lua-0.0.10 --with-ld-opt='-Wl,-rpath,/usr/local/openresty/luajit/lib -Wl,-rpath,/usr/local/openresty/wasmtime-c-api/lib -L/usr/local/openresty/zlib/lib -L/usr/local/openresty/pcre/lib -L/usr/local/openresty/openssl111/lib -Wl,-rpath,/usr/local/openresty/zlib/lib:/usr/local/openresty/pcre/lib:/usr/local/openresty/openssl111/lib' --add-module=/tmp/tmp.itsGpY3Nfs/openresty-1.19.9.1/../mod_dubbo --add-module=/tmp/tmp.itsGpY3Nfs/openresty-1.19.9.1/../ngx_multi_upstream_module --add-module=/tmp/tmp.itsGpY3Nfs/openresty-1.19.9.1/../apisix-nginx-module --add-module=/tmp/tmp.itsGpY3Nfs/openresty-1.19.9.1/../apisix-nginx-module/src/stream --add-module=/tmp/tmp.itsGpY3Nfs/openresty-1.19.9.1/../wasm-nginx-module --add-module=/tmp/tmp.itsGpY3Nfs/openresty-1.19.9.1/../lua-var-nginx-module --with-poll_module --with-pcre-jit --with-stream --with-stream_ssl_module --with-stream_ssl_preread_module --with-http_v2_module --without-mail_pop3_module --without-mail_imap_module --without-mail_smtp_module --with-http_stub_status_module --with-http_realip_module --with-http_addition_module --with-http_auth_request_module --with-http_secure_link_module --with-http_random_index_module --with-http_gzip_static_module --with-http_sub_module --with-http_dav_module --with-http_flv_module --with-http_mp4_module --with-http_gunzip_module --with-threads --with-compat --with-stream --with-http_ssl_module
```
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by jagerzhang**
### Description

Adjust the execution priority of request-id from 11010 to 12015

```
...
  - request-id                     # priority: 12015
  - zipkin                            # priority: 12011
  #- skywalking                  # priority: 12010
  #- opentelemetry            # priority: 12009
...
```

Fixes #7239

### Checklist

- [✔] I have explained the need for this PR and the problem it solves
- [✔] I have explained the changes or the new features added to this PR
- [✔] I have added tests corresponding to this change
- [✖] I have updated the documentation to reflect this change
- [✖] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/opentelemetry.lua [hunk 1]
```diff
@@ -176,7 +176,7 @@ local schema = {
 
 local _M = {
     version = 0.1,
-    priority = 12009,
+    priority = 11009,
```
Comment (tokers) [inline apisix/plugins/opentelemetry.lua:179]: I think it's not a good idea to set it adjacent to the request-id plugin. It's not extendable if we need to insert another plugin between them in the future.

Comment (jagerzhang) [reply apisix/plugins/opentelemetry.lua]: > I think it's not a good idea to set it adjacent to the request-id plugin. It's not extendable if we need to insert another plugin between them in the future.

OK, aready change to 11005

**Comment by tokers** [comment]
@jagerzhang Also, you need to check out the CI failures, we have some test cases to cover the plugin priority settings.

**Comment by jagerzhang** [comment]
@tokers 
CI report this error, I don't know why this file not found.
```
Error: [error] [lua] config_yaml.lua:72: read_apisix_yaml(): failed to fetch /apisix/t/servroot/conf/apisix.yaml attributes: cannot obtain information from file '/apisix/t/servroot/conf/apisix.yaml': No such file or directory
```

**Comment by tokers** [comment]
> @tokers CI report this error, I don't know why this file not found.
> 
> ```
> Error: [error] [lua] config_yaml.lua:72: read_apisix_yaml(): failed to fetch /apisix/t/servroot/conf/apisix.yaml attributes: cannot obtain information from file '/apisix/t/servroot/conf/apisix.yaml': No such file or directory
> ```

Looks like this is the actual error: https://github.com/apache/apisix/runs/6977549863?check_suite_focus=true#step:15:700

**Comment by jagerzhang** [comment]
@tokers I found that the order of plugins was verified in many places, like:
https://github.com/apache/apisix/blob/69bb8b394b8480dd14b518bba5c38970fe7bf24e/t/debug/debug-mode.t#L44-L89
https://github.com/apache/apisix/blob/master/t/admin/plugins.t#L63-L131
https://github.com/apache/apisix/blob/master/conf/config-default.yaml#L335-L410

I think it may be more convenient to uniformly abstract to one place, such as uniformly reading from `config-default yaml`.

**Comment by jagerzhang** [comment]
@tokers 
Hi，What is the cause of this error:
https://github.com/apache/apisix/runs/6995964208?check_suite_focus=true#step:11:52

**Review by spacewander — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Pull request merged**